src/ariel-os-stm32/src/lib.rs

```diff
@@ -80,7 +80,48 @@ pub fn init() -> OptionalPeripherals {
 }
 
 // TODO: find better place for this
+#[allow(clippy::too_many_lines)]
 fn board_config(config: &mut Config) {
+    #[cfg(context = "st-b-l475e-iot01a")]
+    {
+        use embassy_stm32::rcc::*;
+
+        // This board has an LSE clock, we can use it to calibrate the MSI clock
+        config.rcc.ls = LsConfig {
+            rtc: RtcClockSource::LSE,
+            lsi: false,
+            lse: Some(LseConfig {
+                frequency: embassy_stm32::time::Hertz(32768),
+                mode: LseMode::Oscillator(LseDrive::Low),
+            }),
+        };
+        config.rcc.hsi = false;
+
+        config.rcc.msi = Some(MSIRange::RANGE8M);
```

Comment:
The reference manual says the following:

> At 48 MHz, the MSI in PLL-mode can
be used for the USB OTG FS device, saving the need of an external high-speed crystal.

It seems to me like if would be preferable to only use one PLL if possible (and only for SYSCLK) and use the MSI output directly for USB.

```suggestion
        config.rcc.msi = Some(MSIRange::RANGE48M);
```

Reply:
Ah indeed I misunderstood the reference manual.